src/ToolBox/SOS/lldbplugin/services.h

```diff
@@ -252,6 +253,10 @@ class LLDBServices : public ILLDBServices
     // IDebugRegisters
     //----------------------------------------------------------------------------
 
+    HRESULT GetIndexByName(
```

Comment:
ILLDBServices needs to treated like a regular COM interface. Methods can not be added like this. You need to create a ILLDBServices2 interface and put these new methods in that interface.  They can still be implemented in services.cpp but this new ILLDBServices2 needs to be added to the QueryInterface.

On the SOS side (which your changes don't seem to include any usage of this new methods) in the dbgeng.h wrapper, the ILLDBServices2 needs to be QI'ed and if it doesn't exist, they wrapper functions for the IDebugRegisters methods need to fail.